sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/planning/ScanOperationSuite.scala

```diff
@@ -57,7 +57,14 @@ class ScanOperationSuite extends SparkFunSuite {
 
   test("Project which has the same non-deterministic expression with its child Project") {
     val project3 = Project(Seq(colA, colR), Project(Seq(colA, aliasR), relation))
-    assert(ScanOperation.unapply(project3).isEmpty)
+    project3 match {
```

Comment:
I updated `ScanOperation` a little bit to follow `PhyiscalOperation`. Basically, this pattern can always match. The returned plan can be different according to which projects/filters get collected.

Reply:
These tests in `ScanOperationSuite` are not changed in semantics, but just changed for the updated `ScanOperation`?

Reply:
Yes